libs/wordpress-plugin/plugin/trunk/server/activate.php

```diff
@@ -0,0 +1,180 @@
+    $headers = [
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json',
+    ];
+    $data = [
+        'email' => $email,
+        'wordpressInstanceUrl' => get_site_url(),
+        'wordpressSettingsUrl' => $url,
+    ];
+
+    $respStr = wp_remote_post($apiUrl, [
+        'method' => 'POST',
+        'headers' => $headers,
+        'body' => json_encode($data)
+    ])['body'];
+
+    $respJson = json_decode($respStr, true);
+
+    if ($respJson['verificationCodeId']) {
+        block_protocol_update_options([
+            'block_protocol_field_api_email' => $email,
+            'block_protocol_field_api_email_verification_id' => $respJson['verificationCodeId'],
+            'block_protocol_field_api_key' => ''
+        ]);
+    } else if (isset($respJson['errors']) && isset($respJson['errors'][0]) && isset($respJson['errors'][0]['param'])) {
+        $url .= '&' . http_build_query([
+                'invalid_field' => $respJson['errors'][0]['param'],
+                'invalid_field_value' => $respJson['errors'][0]['value'] ?? ''
+            ]);
+    } else {
+        $url .= '&unknown_error=true';
+    }
+
+    exit(wp_redirect($url));
+}
+
+add_action('admin_post_block_protocol_link_by_email', 'block_protocol_link_by_email');
+
+function block_protocol_link_by_api_key()
+{
+    check_admin_referer('block_protocol_link_by_api_key');
+
+    $url = admin_url('admin.php?page=block_protocol');
+
+    if (isset($_POST['key']) && $_POST['key']) {
+        $key = $_POST['key'];
+
+        block_protocol_update_options([
+            'block_protocol_field_api_key' => $key
+        ]);
+        $url .= '&activated=true';
+    }
+
+    exit(wp_redirect($url));
+}
+
+add_action('admin_post_block_protocol_link_by_api_key', 'block_protocol_link_by_api_key');
+
+function block_protocol_activate_submit_button($label, $extra_html = '', $icon = 'arrow')
+{
+    $icons = [
+        'arrow' => '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="fa BPSettingsButtonSvg"><path d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"/></svg>',
+        'tick' => '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="fa BPSettingsButtonSvg"><path d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"/></svg>'
+    ];
+
+    ?>
+    <button type="submit" name="submit"
+            class="BPSettingsButton" <?= $extra_html ?>>
+        <?= $label ?>
+        <?= $icons[$icon]; ?>
+    </button>
+    <?php
+}
+
+function block_protocol_options_page_activate_html()
+{
+    $options = get_option('block_protocol_options');
+    $passedEmailInvalid = $_GET['invalid_field'] === 'email';
```

Comment:
Wonder if we can show a message / console error if the `invalid_field` is `wordpressinstanceurl` – I had this because I was running WP locally but still connected to the prod BP site, and it was rejecting the non-https URL but with no indication of what was going on in the page.

Not a big deal as will only happen in local dev